nvflare/fuel/hci/conn.py

```diff
@@ -27,16 +28,20 @@
 
 LINE_END = "\x03"  # Indicates the end of the line (end of text)
 ALL_END = "\x04"  # Marks the end of a complete transmission (End of Transmission)
-
-
 MAX_MSG_SIZE = 1024
+MAX_DATA_SIZE = 512*1024*1024
+MAX_IDLE_TIME = 60
 
 
 def receive_til_end(sock, end=ALL_END):
     total_data = []
-
+    data_size = 0
+    sock.settimeout(MAX_IDLE_TIME)
     while True:
         data = str(sock.recv(1024), "utf-8")
+        data_size += len(data)
+        if data_size > MAX_DATA_SIZE:
+            raise BufferError("Data size exceeds limit (%d bytes)" % MAX_DATA_SIZE)
```

Comment:
Should use string formatting syntax consistently across NVFLARE: usually ".format" or "f".

Reply:
Fixed